chia/simulator/full_node_simulator.py

```diff
@@ -43,6 +52,26 @@ async def get_all_full_blocks(self) -> List[FullBlock]:
         blocks.reverse()
         return blocks
 
+    async def autofarm_transaction(self, spend_name: bytes32) -> None:
+        if self.auto_farm:
+            self.log.info(f"Autofarm triggered by tx-id: {spend_name.hex()}")
+            new_block = FarmNewBlockProtocol(self.bt.farmer_ph)
+            await self.farm_new_transaction_block(new_block)
+
+    async def update_autofarm_config(self, enable_autofarm: Optional[bool] = None) -> bool:
```

Comment:
we should not pass in Optional, it's confusing. Better to pass in a bool, and check if None before calling this method